cluster/nginx.go

```diff
@@ -294,31 +337,154 @@ func UpdateNginxConfiguration(ctx *Context) chan error {
+		if accessTenantResult.Error != nil {
+			log.Println(accessTenantResult.Error)
+			continue
 		}
+		accessTenant := accessTenantResult.AccessKeyToTenantShortName
 
-		err = <-ReDeployNginxResolver(ctx)
-		if err != nil {
-			ch <- err
-			return
+		mapLine := `			"%s" "%s";
+`
+		destinationMapping.WriteString(fmt.Sprintf(mapLine, accessTenant.AccessKey, accessTenant.TenantShortName))
+	}
+
+	var nginxConfiguration bytes.Buffer
+	nginxConfiguration.WriteString(`
+			user nginx;
+			worker_processes auto;
+			error_log /dev/stdout debug;
+			pid /var/run/nginx.pid;
+
+			events {
+				worker_connections  1024;
+			}`)
+
+	nginxConfiguration.WriteString(fmt.Sprintf(`
+			http {
+			upstream portalproxy {
+				server portal-proxy:80;
+			}
+			upstream tenancy {
+				server portal-proxy:80;
+			}
+
+			%s
+		
+			map $http_authorization $access_destination {
+			default               "";
+				"~*Credential=(?<access_key>.*?)\/" "$access_key";
+			}
+		
+			# map to different upstream backends based on header
+			map $access_destination $pool {
+				"" "portalproxy";
+`, tenantUpstreams.String()))
+
+	nginxConfiguration.WriteString(destinationMapping.String())
+	appDomain := getS3Domain()
+	nginxConfiguration.WriteString(fmt.Sprintf(`
+			}
+
+
+			log_format  main  '$http_host - $remote_addr - $remote_user [$time_local] "$request" '
+																		'$status $body_bytes_sent "$http_referer" '
+																		'"$http_user_agent" "$http_x_forwarded_for"';
+				server {
+					access_log /var/log/nginx/access.log main;
+					location / {
+						 proxy_set_header Upgrade $http_upgrade;
+						 proxy_set_header Connection "upgrade";
+						 client_max_body_size 0;
+						 proxy_set_header Host $http_host;
+						 proxy_set_header X-Real-IP $remote_addr;
+						 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+						 proxy_set_header X-Forwarded-Proto $scheme;
+						 proxy_set_header X-Frame-Options SAMEORIGIN;
+						 proxy_buffers 256 16k;
+						 proxy_buffer_size 16k;
+						 client_body_timeout 60;
+						 send_timeout 300;
+						 lingering_timeout 5;
+						 proxy_connect_timeout 90;
+						 proxy_send_timeout 300;
+						 proxy_read_timeout 90s;
+						 proxy_pass http://portal-proxy:80;
+					}
+				}
+
+				server {
+					server_name %s;
+					ignore_invalid_headers off;
+					client_max_body_size 0;
+					proxy_buffering off;
+					location / {
+						proxy_http_version 1.1;
+						proxy_set_header Host $http_host;
+						proxy_read_timeout 15m;
+						proxy_request_buffering off;
+						proxy_pass http://$pool;
+					}
+				}
+		`, appDomain))
+
+	bucketRoutes := streamBucketToTenantServices()
+	for bucketRouteResult := range bucketRoutes {
+		if bucketRouteResult.Error != nil {
+			log.Println(bucketRouteResult.Error)
+			continue
 		}
+		bucketRoute := bucketRouteResult.BucketToService
+		serverBlock := fmt.Sprintf(`
+				server {
+					server_name %s.%s;
+					ignore_invalid_headers off;
+					client_max_body_size 0;
```

Comment:
0 means no limit in the body size? if yes I think we should declare a maximum limit, even if this is big , to avoid potential DOS

Reply:
but now we are passing files, we could set it to MinIO max which is 5TB